lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -657,8 +657,12 @@ sub overview {
         groups => $groups,
         until => $until,
     );
-    my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
-    ($stash{archs}, $stash{results}, $stash{aggregated}) = $self->prepare_job_results(\@latest_jobs);
+    my $limit = $search_args->{limit} = OpenQA::App->singleton->config->{misc_limits}->{tests_overview_max_jobs};
+    my $jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args);
+    my $limit_exceeded = $jobs->count >= $limit ? $limit : undef;
```

Comment:
Not sure how expensive the count here is (additionally to the real select).
Why not do a select with a limit?

Reply:
I suppose the select *is* limited. That's also the reason for the `>=` (and not just `>`).

Reply:
Turned out to be really cheap.